From c90b056c902540c2d65ed664f64b7fba4b9a7e68 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andr=C3=A9=20Erdmann?= <dywi@mailerd.de>
Date: Mon, 25 Aug 2014 23:41:28 +0200
Subject: [PATCH 1/6] tlp-functions, get_disk_dev(): minor fix

Small fix for the unlikely case that the disk id $1
ends with a partition number >= 10 (contains one or more '0' chars).

Another (slightly less reliable) solution would be
the shell's builtin variable substitution:

  > disk_dev=${disk_id%%-part[0-9]*}
  ...
  > disk_dev=${disk_dev%%[0-9]*}
---
 tlp-functions | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tlp-functions b/tlp-functions
index fa216cd..b3e5130 100755
--- a/tlp-functions
+++ b/tlp-functions
@@ -634,7 +634,7 @@ get_disk_dev () { # translate disk id to device (sdX)
     if [ -h /dev/disk/by-id/$1 ]; then
         # $1 is disk id
         disk_id=$1
-        disk_dev=$(echo $disk_id | sed -r 's/-part[1-9]+$//')
+        disk_dev=$(echo $disk_id | sed -r 's/-part[1-9][0-9]*$//')
         disk_dev=$(readlink /dev/disk/by-id/$disk_dev)
         disk_dev=${disk_dev##*/}
     else
@@ -643,7 +643,7 @@ get_disk_dev () { # translate disk id to device (sdX)
         disk_id=""
     fi
     # strip partition number
-    disk_dev=$(echo $disk_dev | sed -r 's/[1-9]+$//')
+    disk_dev=$(echo $disk_dev | sed -r 's/[1-9][0-9]*$//')
 }
 
 set_disk_apm_level () { # set disk apm level
-- 
2.0.4

